integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/glue/steps/mocks/OslpDeviceSteps.java

```diff
@@ -1250,4 +1423,17 @@ private static String convertIpAddress(final ByteString byteString) {
         final String ipValue = stringBuilder.toString();
         return ipValue.substring(0, ipValue.length() - 1);
     }
+
+    private Map<String, String> setDeviceUid(final Map<String, String> settings, final String deviceUid) {
+        settings.put(PlatformPubliclightingKeys.KEY_DEVICE_UID, deviceUid);
+        return settings;
+    }
+
+    private String getDeviceUID(Map<String, String> settings) {
+        if (settings == null) {
+            settings = new HashMap<>();
```

Comment:
Maybe nicer to just return the default DEVICE_UID instead --> That's what it does. I use this method as helper method to get rid of this code that is used in may places (which this method encapsulates) 
       
return getString(settings, PlatformPubliclightingKeys.KEY_DEVICE_UID,
                PlatformPubliclightingDefaults.DEVICE_UID);

Reply:
Regarding just returning the default device UID, that is _not_ what it does. More precisely, that is not what the line a commented on does, it _is_ what the method does, but that wasn't my point.
What it does, is producing some return value by assigning to a parameter (bad practice), which causes a side effect later-on.
The method is rather short, so it is not that hard to comprehend, and you can leave it as is if you like it better this way.

Reply:
Oh, agreed, thanks for pointing out. It can potentially hide an unwanted edge case (cq bug) where settings == null
I'll remove the null check since it is/should not be needed anyway.